newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -485,6 +488,25 @@ export default class AuthenticatedUserProvider extends React.Component<
     }
   };
 
+  // Put user cloud projects listing promise in common to prevent duplicate api call:
+  // - First one comes from user authenticating and automatically fetching
+  //   their cloud projects;
+  // - Second one comes from the homepage fetching the cloud projects regularly.
```

Comment:
As the comment implies, this create more complexity in this file.
It's a good idea to de-duplicate API calls like this! But this should probably be handled by something designed for it.

Could we have something like: `_listUserCloudProjects = deduplicateApiCall(listUserCloudProjects)`? 
A bit of tricky typing, but once it's set up, it could in the future almost be a hook or a helper used in other places.